### ISSUE TITLE
add portable nproc shim

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#415]
People keep running into this portability issue.

related to #167  and PR #411 

https://github.com/oilshell/oil/pull/194/files

I think this is a good place to start our `sh-env` ... i.e. insert a script called `nproc` in the $PATH, and then insert that on non-coreutils OSes.

**Comment by andychu** [comment]
cc @jyn514

**Comment by jyn514** [comment]
What if we put these in `bin` and appended `bin` to $PATH? If `nproc` is already installed, that gets run, otherwise our shim gets run. That means we don't have to check if it's installed as well, if the shim gets run we know that it's not there.

**Comment by jyn514** [comment]
I guess that doesn't work for things like `make` where both `make` and `gmake` can be installed.

**Comment by andychu** [comment]
Yup that's the idea.  Some might be at the beginning of $PATH, and some might be at the end.

I still have to work out the exact code structure, but I think this is the way forward.

It's not very different from the patch you sent, but it's just organizing the code in a slightly different way.

**Comment by andychu** [comment]
For make, we can have `sh-env/linux/make` and `sh-env/freebsd/make`, the latter of which calls out to `gmake`.  And then that has to be on the front of the path, not the end, so that the system `make` doesn't get called.

**Comment by andychu** [comment]
BTW if something is blocking you on doing anything with FreeBSD, let me know.  I'd be curious to know if the next release tarball can be built on FreeBSD, provided we have the `configure` change.

Do the instructions have to be changed to run `gmake` on FreeBSD ?

Long term I want to get rid of any make, not just GNU make, but will take awhile.

**Comment by jyn514** [comment]
`master` builds without changes on FreeBSD. It builds with `--readline=/path/to/readline` with the patch to `expr`.

**Comment by andychu** [comment]
note: spec tests now allow `MAX_PROCS=1` which sort of sidesteps this issue... that's how they run in the continuous build, and it's not even that slow..

https://github.com/oilshell/oil/blob/master/test/spec.sh#L241
